cms/toolbar/utils.py

```diff
@@ -117,14 +120,41 @@ def get_toolbar_from_request(request):
     return getattr(request, 'toolbar', EmptyToolbar(request))
 
 
+def get_querystring_modifier(obj, url, language=None):
+    """
+    Get CMS_ENDPOINT_LIVE_URL_QUERYSTRING_PARAM as the parameter, and the live url, if the object is
+    PageContent
+    """
+    url_param = getattr(settings, 'CMS_ENDPOINT_LIVE_URL_QUERYSTRING_PARAM', None)
+    if not url_param:
+        raise ImproperlyConfigured(
+            """
+            With 'CMS_ENDPOINT_LIVE_URL_QUERYSTRING_PARAM_ENABLED' set to True,
+            CMS_ENDPOINT_LIVE_URL_QUERYSTRING_PARAM must be provided
+            """
+        )
+    if not isinstance(obj, PageContent):
```

Comment:
Could you add a test for this - so when the setting is enabled, but the object is not `PageContent`, the url is returned unchanged?

Reply:
No problem!